packages/quantic/force-app/main/default/lwc/quanticTabBar/__tests__/quanticTabBar.test.js

```diff
@@ -0,0 +1,123 @@
+  mockSlotAssignedNodes(assignedElements);
+
+  const element = createElement('c-quantic-tab-bar', {
+    is: QuanticTabBar,
+  });
+  for (const [key, value] of Object.entries(options)) {
+    element[key] = value;
+  }
+
+  document.body.appendChild(element);
+  return element;
+}
+
+// Helper function to wait until the microtask queue is empty.
+function flushPromises() {
+  return new Promise((resolve) => setTimeout(resolve, 0));
+}
+
+function cleanup() {
+  // The jsdom instance is shared across test cases in a single file so reset the DOM
+  while (document.body.firstChild) {
+    document.body.removeChild(document.body.firstChild);
+  }
+  jest.clearAllMocks();
+}
+
+describe('c-quantic-tab-bar', () => {
+  afterEach(() => {
+    cleanup();
+  });
+
+  const exampleSlots = [
+    {
+      engineId: 'example engine id',
+      label: 'Item One',
+      expression: 'example expression',
+      isActive: true,
+    },
+    {
+      engineId: 'example engine id',
+      label: 'Item Two',
+      expression: 'example expression',
+      isActive: false,
+    },
+  ];
+
+  const exampleAssignedElements = exampleSlots;
+
+  it('should display all the tabs without displaying the dropdown list', async () => {
+    const expectedOpenDropdownClass = 'slds-is-open';
+
+    const element = createTestComponent(
+      defaultOptions,
+      exampleAssignedElements
+    );
+    await flushPromises();
+
+    exampleSlots.forEach((tabItem, index) => {
+      expect(tabItem.label).toEqual(exampleSlots[index].label);
+      expect(tabItem.engineId).toEqual(exampleSlots[index].engineId);
+      expect(tabItem.expression).toEqual(exampleSlots[index].expression);
+      expect(tabItem.isActive).toEqual(exampleSlots[index].isActive);
+    });
+
+    const dropdown = element.shadowRoot.querySelector(selectors.dropdown);
+    expect(dropdown.classList).not.toContain(expectedOpenDropdownClass);
+  });
+
+  describe('when the light theme property is set to true', () => {
+    it('should display the component with the light theme styles', async () => {
+      const expectedDarkThemeClass = 'slds-theme_shade';
+      const element = createTestComponent({lightTheme: true});
+      await flushPromises();
+
+      const tabBarContainer = element.shadowRoot.querySelector(
+        selectors.tabBarContainer
+      );
+      expect(tabBarContainer.classList).not.toContain(expectedDarkThemeClass);
+    });
+  });
+
+  describe('when the light theme property is set to false', () => {
+    it('should display the component with the light theme styles', async () => {
+      const expectedDarkThemeClass = 'slds-theme_shade';
+      const element = createTestComponent({lightTheme: false});
+      await flushPromises();
+
+      const tabBarContainer = element.shadowRoot.querySelector(
+        selectors.tabBarContainer
+      );
+      expect(tabBarContainer.classList).toContain(expectedDarkThemeClass);
+    });
+  });
```

Comment:
I don't know if you agree but I find we always have too many `describe` xD

```suggestion
  describe('the lightTheme property', () => {
    it('when true, should display the component as light theme', async () => {
      const expectedDarkThemeClass = 'slds-theme_shade';
      const element = createTestComponent({lightTheme: true});
      await flushPromises();

      const tabBarContainer = element.shadowRoot.querySelector(
        selectors.tabBarContainer
      );
      expect(tabBarContainer.classList).not.toContain(expectedDarkThemeClass);
    });

    it('when false, should display the component as dark theme', async () => {
      const expectedDarkThemeClass = 'slds-theme_shade';
      const element = createTestComponent({lightTheme: false});
      await flushPromises();

      const tabBarContainer = element.shadowRoot.querySelector(
        selectors.tabBarContainer
      );
      expect(tabBarContainer.classList).toContain(expectedDarkThemeClass);
    });
  });
```